espnet2/train/trainer.py

```diff
@@ -241,7 +242,11 @@ def run(
                         else None
                     ),
                     find_unused_parameters=trainer_options.unused_parameters,
+                    static_graph=True,
```

Comment:
I think `gradient_as_bucket_view` can be left as default, but `static_graph` and ` dp_model.register_comm_hook(None, bf16_compress_hook)` may need to be configurable flags by the user. Any suggestions?